libs/scalajslib/worker/1/src/mill/scalajslib/worker/ScalaJSWorkerImpl.scala

```diff
@@ -22,12 +26,7 @@ import org.scalajs.jsenv.{Input, JSEnv, RunConfig}
 import org.scalajs.testing.adapter.TestAdapter
 import org.scalajs.testing.adapter.TestAdapterInitializer as TAI
 
-import scala.collection.mutable
-import scala.ref.SoftReference
-import com.armanbilge.sjsimportmap.ImportMappedIRFile
-import mill.constants.InputPumper
-
-class ScalaJSWorkerImpl extends ScalaJSWorkerApi {
+class ScalaJSWorkerImpl(jobs: Int) extends ScalaJSWorkerApi {
```

Comment:
Having the jobs in the ctr means, we need to drop all cached linkers (since we need to recreate the instance) when the jobs config changes, which looks wasteful to me. But after a closer look, it looks the `jobs` value is already a ctr-arg of the `ScalaJSWorker`, which already suffers from the same issue. So, this PR looks ok to me, but we probably should overhaul the design .